internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -498,14 +487,154 @@ func resourceApiManagementApiCreateUpdate(d *pluginsdk.ResourceData, meta interf
 		params.Properties.TermsOfServiceURL = pointer.To(v.(string))
 	}
 
-	if err := client.CreateOrUpdateThenPoll(ctx, newId, params, api.CreateOrUpdateOperationOptions{IfMatch: pointer.To("*")}); err != nil {
+	if err := client.CreateOrUpdateThenPoll(ctx, id, params, api.CreateOrUpdateOperationOptions{IfMatch: pointer.To("*")}); err != nil {
 		return fmt.Errorf("creating/updating %s: %+v", id, err)
 	}
 
 	d.SetId(id.ID())
 	return resourceApiManagementApiRead(d, meta)
 }
 
+func resourceApiManagementApiUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.ApiClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	path := d.Get("path").(string)
+	version := d.Get("version").(string)
+	versionSetId := d.Get("version_set_id").(string)
+	displayName := d.Get("display_name").(string)
+	protocolsRaw := d.Get("protocols").(*pluginsdk.Set).List()
+	protocols := expandApiManagementApiProtocols(protocolsRaw)
+	sourceApiId := d.Get("source_api_id").(string)
+	serviceUrl := d.Get("service_url").(string)
+
+	if version != "" && versionSetId == "" {
+		return errors.New("setting `version` without the required `version_set_id`")
+	}
+
+	if sourceApiId == "" && (displayName == "" || protocols == nil || len(*protocols) == 0) {
+		return errors.New("`display_name`, `protocols` are required when `source_api_id` is not set")
+	}
+
+	id, err := api.ParseApiID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	apiType := api.ApiTypeHTTP
+	if v, ok := d.GetOk("api_type"); ok {
+		apiType = api.ApiType(v.(string))
+	}
+	soapApiType := soapApiTypeFromApiType(apiType)
+
+	// If import is used, we need to send properties to Azure API in two operations.
+	// First we execute import and then updated the other props.
+	if d.HasChange("import") {
+		if vs, hasImport := d.GetOk("import"); hasImport {
+			apiParams := expandApiManagementApiImport(vs.([]interface{}), apiType, soapApiType,
+				path, serviceUrl, version, versionSetId)
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, apiParams, api.CreateOrUpdateOperationOptions{}); err != nil {
+				return fmt.Errorf("creating/updating %s: %+v", *id, err)
+			}
+		}
+	}
+
+	prop := &api.ApiCreateOrUpdateProperties{
```

Comment:
`TestAccApiManagementApi_importUpdate` is failing because we're building a new payload rather than using the existing one.